### PR TITLE
Feature: do dev environment via makefile

### PR DESCRIPTION
**Pull request by markelog**
Simplifies dev environment creation. Will also planing to utilize
this logic for the LDAP benchmarking

### CODE REVIEW

**Comment by bergquist** [comment]
I dont know that much about makefiles. But should this be placed in a new makefile in `devenv`?

**Comment by xlson** [comment]
@markelog How is it supposed to be used?

File: Makefile [hunk 1]
```diff
@@ -84,6 +84,18 @@ revive: scripts/go/bin/revive
 		-config ./scripts/go/configs/revive.toml \
 		$(GO_FILES)
 
+# create docker-compose file with provided sources and start them
+env: env-down
```
Comment (xlson) [inline Makefile]: How is this supposed to be used? Not sure if `env actually need to depend on `env-down`, `docker-compose up` should take care of removing cointainers as well as adding them unless I'm mistaken.

Comment (markelog) [reply Makefile]: Added an example as makefile comment  – 

`$ make env sources=postgres,openldap`

> Not sure if env actually need to depend on env-down, docker-compose up` should take care of removing cointainers as well as adding them unless I'm mistaken.

Just tried it - If you execute `docker-compose up` its enters the shell of the containers defined in `docker-compose.yaml` it does not extinguish already running containers that *were* defined in `docker-compose.yaml` but not anymore

**Comment by markelog** [comment]
@bergquist 

I guess it depends on what you want to achive. 

Like if you want to execute these `make` commands from `devenv` folder. Or if you would prefer to include potential `devenv` makefile to the main one.

Latter, in my experience, is used when makefile becomes complicated enough to divide it on "modules". 

For me, right now, it's more like `npm` scripts at the moment so it's nice to execute those commands from root

File: Makefile (hunk 1 above)
Comment (bergquist) [reply Makefile]: `docker-compose down` solved broken docker-compose setup for me a bunch of times. I would not mind depending on it to reduce errors on `up`

File: Makefile [hunk 2]
```diff
@@ -84,6 +84,19 @@ revive: scripts/go/bin/revive
 		-config ./scripts/go/configs/revive.toml \
 		$(GO_FILES)
 
+# create docker-compose file with provided sources and start them
+# example: make env sources=postgres,openldap
+env: env-down
```
Comment (bergquist) [inline Makefile]: I wonder if `env` is a good name. Would `devenv` work or would that conflict with the folder name? I need to read up on makefiles :)

Comment (markelog) [reply Makefile]: We can define it as `devenv`! We would just have defined as `PHONY` :)

Comment (bergquist) [reply Makefile]: Then I think we should go with `devenv` since `env` is a common word that can have many different meanings.

Comment (markelog) [reply Makefile]: Done!

**Review by bergquist — APPROVED** [review]
LGTM!

File: Makefile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 -include local/Makefile
 
-.PHONY: all deps-go deps-js deps build-go build-server build-cli build-js build build-docker-dev build-docker-full lint-go test-go test-js test run clean gosec revive
+.PHONY: all deps-go deps-js deps build-go build-server build-cli build-js build build-docker-dev build-docker-full lint-go test-go test-js test run clean gosec revive env env-down
```
Comment (xlson) [inline Makefile]: Need to replace env and env-down with devenv and devenv-down, same below for the devend dependency.

**Review by xlson — APPROVED** [review]
Great work. Let's merge this after the env and env-down references have been fixed.

**Pull request merged**